packages/cloudflare/src/api/do-sharded-tag-cache.ts

```diff
@@ -27,46 +34,159 @@ interface ShardedD1TagCacheOptions {
    * @default 5
    */
   regionalCacheTtlSec?: number;
+
+  /**
+   * Whether to enable shard replication
+   * Shard replication will duplicate each shards into N replicas to spread the load even more
+   * For example with N being 2, tag `tag1` could be read from 2 different durable object instance
+   * On read you only need to read from one of the shards, but on write you need to write to all shards
+   * @default false
+   */
+  enableShardReplication?: boolean;
+
+  /**
+   * The number of replicas that will be used for shard replication
+   * Soft shard replicas are more often accessed than hard shard replicas, so it is recommended to have more soft replicas than hard replicas
+   * Soft replicas are for internal next tags used for `revalidatePath` (i.e. `_N_T_/layout`, `_N_T_/page1`), hard replicas are the tags defined in your app
+   * @default { softShards: 4, hardShards: 2 }
+   */
+  shardReplicationOptions?: {
+    numberOfSoftReplicas: number;
+    numberOfHardReplicas: number;
+  };
+
+  /**
+   * The number of retries to perform when writing tags
+   * @default 3
+   */
+  maxWriteRetries?: number;
 }
-class ShardedD1TagCache implements NextModeTagCache {
+
+interface TagCacheDOIdOptions {
+  baseShardId: string;
+  numberOfReplicas: number;
+  shardType: "soft" | "hard";
+  replicaId?: number;
+}
+export class TagCacheDOId {
+  baseShardId: string;
```

Comment:
Should we call "base-soft/hard" the shard?

```suggestion
  shardId: string;
```